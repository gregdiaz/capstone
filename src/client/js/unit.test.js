const {GeonamesEntry} = require('./ui');

describe('GeonamesEntry',  () => {
    test('should return an empty array', () => {
        const expected = GeonamesEntry([]);

        expect(expected).toBeInstanceOf(Array);
        expect(expected.length).toEqual(0);
    });

    test('should return and empty array when undefined is passed', () => {
        const expected = GeonamesEntry(undefined);

        expect(expected).toBeInstanceOf(Array);
        expect(expected.length).toEqual(0);
    });

    test('should return an array with one element', () => {
        const data = [
            {
                countryCode: 'test',
                lat: 11.2341324,
                lng: 12.3003408
            }
        ]
        const expected = GeonamesEntry(data);

        expect(expected).toBeInstanceOf(Array);
        expect(expected.length).toEqual(data.length);
        expect(expected[0]).toMatchSnapshot();
    });

});

