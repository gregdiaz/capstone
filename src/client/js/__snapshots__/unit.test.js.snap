// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeonamesEntry should return an array with one element 1`] = `
<div
  class="entry"
>
  <p />
  <p />
  <p />
</div>
`;
